scripts/generate-discord-public-report.bash

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+set -e
+
+settlement_collection_file="$1"
+
+if [[ -z $settlement_collection_file ]]
+then
+    echo "Usage: $0 <settlement collection file>" >&2
+    exit 1
+fi
+
+epoch="$(<"$settlement_collection_file" jq '.epoch' -r)"
+if (( $(<"$settlement_collection_file" jq '.settlements | length' -r) == 0 ))
+then
+    echo "No settlements in epoch $epoch."
+    exit
+fi
+
+decimal_format="%0.9f"
+
+function fmt_human_number {
+    numfmt --to si $@
+}
+export -f fmt_human_number
+
+echo "Total settlements in epoch $epoch: ☉$(<"$settlement_collection_file" jq '[.settlements[].claims_amount / 1e9] | add' | xargs printf $decimal_format)"
+echo
+echo "                                vote account    settlement                   reason   stake     funded by"
+echo "--------------------------------------------+-------------+------------------------+-------+-------------"
+while read -r settlement
+do
+    vote_account=$(<<<"$settlement" jq '.vote_account' -r)
+    claims_amount=$(<<<"$settlement" jq '.claims_amount / 1e9' -r | xargs printf $decimal_format)
+    protected_stake=$(<<<"$settlement" jq '[.claims[].active_stake] | add / 1e9' -r | xargs -I{} bash -c 'fmt_human_number "$@"' _ {})
+    reason_code=$(<<<"$settlement" jq '.reason | keys[0]' -r)
+
+    if ! [[ $reason_code == "ProtectedEvent" ]]
```

Comment:
I'm not clear if that's worthy but somehow I feel it would be nice to have the "constant" somehow documented. I mean in the Rust code to have some note at the struct definition that's the name is used in pipeline bash script?
Does it sound reasonable or not?

Reply:
It does make sense, at the same time pretty much all fields of all collections are already used by other processes after those collections are stored in the buckets. The names should not change - we will have to document all places which would be broken by this...